test/linalg/generic.jl

```diff
@@ -89,42 +95,76 @@ y = ['a','b','c','d','e']
 
 @test trace(Bidiagonal(ones(5),zeros(4),true)) == 5
 
-# 2-argument version of scale
-a = reshape([1.:6;], (2,3))
-@test scale(a, 5.) == a*5
-@test scale(5., a) == a*5
-@test scale(a, [1.; 2.; 3.]) == a.*[1 2 3]
-@test scale([1.; 2.], a) == a.*[1; 2]
-@test scale(a, [1; 2; 3]) == a.*[1 2 3]
-@test scale([1; 2], a) == a.*[1; 2]
-@test scale(eye(Int, 2), 0.5) == 0.5*eye(2)
-@test scale([1; 2], sub(a, :, :)) == a.*[1; 2]
-@test scale(sub([1; 2], :), a) == a.*[1; 2]
-@test_throws DimensionMismatch scale(a, ones(2))
-@test_throws DimensionMismatch scale(ones(3), a)
-
-# 2-argument version of scale!
-@test scale!(copy(a), 5.) == a*5
-@test scale!(5., copy(a)) == a*5
-b = randn(Base.LinAlg.SCAL_CUTOFF) # make sure we try BLAS path
-@test scale!(copy(b), 5.) == b*5
-@test scale!(copy(a), [1.; 2.; 3.]) == a.*[1 2 3]
-@test scale!([1.; 2.], copy(a)) == a.*[1; 2]
-@test scale!(copy(a), [1; 2; 3]) == a.*[1 2 3]
-@test scale!([1; 2], copy(a)) == a.*[1; 2]
-@test_throws DimensionMismatch scale!(a, ones(2))
-@test_throws DimensionMismatch scale!(ones(3), a)
-
-# 3-argument version of scale!
-@test scale!(similar(a), 5., a) == a*5
-@test scale!(similar(a), a, 5.) == a*5
-@test scale!(similar(a), a, [1.; 2.; 3.]) == a.*[1 2 3]
-@test scale!(similar(a), [1.; 2.], a) == a.*[1; 2]
-@test scale!(similar(a), a, [1; 2; 3]) == a.*[1 2 3]
-@test scale!(similar(a), [1; 2], a) == a.*[1; 2]
-@test_throws DimensionMismatch scale!(similar(a), a, ones(2))
-@test_throws DimensionMismatch scale!(similar(a), ones(3), a)
-@test_throws DimensionMismatch scale!(Array(Float64, 3, 2), a, ones(3))
+
+# array and subarray tests
+let aa = reshape([1.:6;], (2,3))
+    for atype in ("Array", "SubArray")
```

Comment:
Minor detail, but maybe just use the types here instead of strings, i.e. `Array` and `SubArray`